lib/services/search.js

```diff
@@ -0,0 +1,404 @@
+ * @param {string} type
+ * @param {string} ref, e.g. 'localhost.dev.nymag.biz/scienceofus/components/article/instances/section-test'
+ * @param {object} source
+ * @returns {Promise}
+ */
+function put(index, type, ref, source) {
+  return client.index({
+    index: index,
+    type: type,
+    id: ref,
+    body: source
+  }).then(function (resp) {
+    log('info', resp);
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Delete an Elasticsearch document
+ *
+ * @param {string} index
+ * @param {string} type
+ * @param {string} ref, e.g. 'localhost.dev.nymag.biz/scienceofus/components/article/instances/section-test'
+ * @returns {Promise}
+ */
+function del(index, type, ref) {
+  return client.delete({
+    index: index,
+    type: type,
+    id: ref
+  }).then(function (resp) {
+    log('info', resp);
+    return resp;
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Perform multiple index operations
+ *
+ * @param {Array} ops
+ * @returns {Promise}
+ */
+function batch(ops) {
+  return client.bulk({
+    body: ops
+  }).then(function (resp) {
+    if (resp && resp.errors === true) {
+      let str = ['Client.bulk errored on ' + util.inspect(ops, {depth: 10})];
+
+      return bluebird.reject(new Error(str));
+    }
+  });
+}
+
+/**
+ * Create the ES Client or an empty object
+ *
+ * @param {Object} overrideClient
+ */
+function setup(overrideClient) {
+  if (!module.exports.endpoint && !overrideClient) {
+    throw new Error('No Elastic endpoint or client override');
+  }
+
+  // Set the exported client
+  module.exports.client = client = module.exports.endpoint && !overrideClient ? new elastic.Client(serverConfig) : _.assign(module.exports.client, overrideClient);
+}
+
+/**
+ * Sets the alias suffix for use
+ * @param {[type]} suffix [description]
+ */
+function setAliasSuffix(suffix) {
+  if (_.isString(suffix)) {
+    aliasSuffix = suffix;
+  } else {
+    throw new Error('Alias suffix required to be a string');
+  }
+}
+
+module.exports.setup = setup;
+module.exports.setAliasSuffix = setAliasSuffix;
+module.exports.endpoint = endpoint;
+module.exports.healthCheck = healthCheck;
+module.exports.initIndex = initIndex;
+module.exports.initAlias = initAlias;
+module.exports.delIndex = delIndex;
+module.exports.createIndexIfNone = createIndexIfNone;
+module.exports.createAliasIfNone = createAliasIfNone;
+module.exports.existsAlias = existsAlias;
+module.exports.existsIndex = existsIndex;
+module.exports.initMapping = initMapping;
+module.exports.existsMapping = existsMapping;
+module.exports.createMappingIfNone = createMappingIfNone;
+module.exports.createIndexName = createIndexName;
+module.exports.convertRedisBatchtoElasticBatch = convertRedisBatchtoElasticBatch;
+module.exports.del = del;
+module.exports.batch = batch;
+module.exports.put = put;
+module.exports.query = query;
```

Comment:
In general, I think I'd like to see assurance that index name and alias can never be the same. If they ever are you could generate errors on some of these ops.  Currently we do that by using simple versioning on index name (e.g. append "_v1"), so I'd like to see a naming standard proposed that is checked before any create/update index operation is called.